js/supermarkets.js

```diff
@@ -18,27 +19,25 @@ function initMap() {
 searchMarketsNear('München, Deutschland');
 
 /*Triggering Button when Enter is pressed */
-function triggerBtn() {
-    input = document.getElementById('inputValue');
-    input.addEventListener("keyup", function(event) {
-        if (event.keyCode === 13) {
-            event.preventDefault();
-            document.getElementById("myBtn").click();
+function triggerFunction() {
+    input.addEventListener("keyup", function(e) {
+        if (e.key === 'Enter') {
+            searchMarketsNear(input.value);
         }
     });
 }
 
 /*Searching markets by ZIP code or region */
 function searchMarketsNear(searchedPlace) {
-    input = document.getElementById('inputValue').value;
-    if (input !== null && input !== '') {
-        searchedPlace = input;
+    inputForSearch = input.value;
+    if (inputForSearch !== '') {
+        searchedPlace = inputForSearch;
     } else {
         searchedPlace = 'München, Deutschland';
```

Comment:
this line will always be executed when your input is empty. So the parameter which you can pass inside this function will never be used as a value for search supermarkets.